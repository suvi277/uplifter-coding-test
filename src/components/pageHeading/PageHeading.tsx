import React from 'react';

interface Props {
    title: string;
}

const PageHeading: React.FC<Props> = ({ title }) => {
    return <h1 className="pt-4 text-center text-uppercase">{title}</h1>;
};
export default PageHeading;
