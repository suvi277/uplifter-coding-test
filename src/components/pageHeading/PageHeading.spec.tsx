import React from 'react';
import { shallow, mount } from 'enzyme';
import PageHeading from './PageHeading';

describe('PageHeading Component', () => {
    const wrapper = shallow(<PageHeading title="" />);
    it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should exist h1 tag', () => {
        expect(wrapper.find('h1')).toHaveLength(1);
    });

    it('should match h1 innerText with prop title ', () => {
        const pageHeadingComp = mount(<PageHeading title="Some Title" />);

        expect(pageHeadingComp.find('h1').render().text()).toEqual('Some Title');
    });
});
