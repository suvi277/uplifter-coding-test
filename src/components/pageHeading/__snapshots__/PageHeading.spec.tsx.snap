// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading Component renders as expected 1`] = `ShallowWrapper {}`;
